### DISCUSSION TITLE
Run getServerSideProps on server using next/router.

### DISCUSSION THREAD

**Discussion by AbdullahAnsarii** [vercel/next.js#38680] [Ideas]
### Describe the feature you'd like to request

I'd like an option in `next/router` and `next/link` to always run `getServerSideProps` on the server.

### Describe the solution you'd like

I'd like a property in `router.push(url, as, options)` more specifically in `options` which tells the page to run `getServerSideProps` on server.

### Describe alternatives you've considered

I've used `window.location.replace` in javascript, and it works as expected.

**Comment by icyJoseph** [comment]
Hi,

You can use `router.replace(router.asPath)` to re-trigger GSSP. Here's an article covering this in depth, https://www.joshwcomeau.com/nextjs/refreshing-server-side-props/.

```tsx
import { useRouter } from 'next/router';

function SomePage(props) {
  const router = useRouter();
  // Call this function whenever you want to
  // refresh props!
  const refreshData = () => {
    router.replace(router.asPath);
  }
}

export async function getServerSideProps(context) {
  // Database logic here
}
```

For clarity, `getServerSideProps`, GSSP, always runs on the server.

**Comment by AbdullahAnsarii** [comment]
Thanks, @icyJoseph. It's clear to me now.
